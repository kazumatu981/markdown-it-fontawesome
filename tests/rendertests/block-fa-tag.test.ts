import { describe, test, expect } from '@jest/globals';
import MarkdownIt from 'markdown-it';
import { BlockFaTagTest } from './snapshot-testcases';

const markdownItFontawesome = require('../../index');

function render(markdown: string): string {
    const md = new MarkdownIt();
    md.use(markdownItFontawesome);

    return md.render(markdown);
}

describe('Inline FaTag Tests.', () => {
    BlockFaTagTest.forEach((testCase) => {
        test(testCase.description, () => {
            const actual = render(testCase.markdown);
            expect(actual).toMatchSnapshot();
        });
    });
});
