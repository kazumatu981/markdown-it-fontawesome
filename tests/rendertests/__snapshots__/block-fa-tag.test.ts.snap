// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inline FaTag Tests. very simple tag in a list' 1`] = `
"<ul class="fa-ul">
<li><span class="fa-li"><i class="fa-user"></i></span> test001</li>
<li><span class="fa-li"><i class="fa-user"></i></span> test002</li>
<li><span class="fa-li"><i class="fa-user"></i></span> test003</li>
</ul>
"
`;
